src/inmanta/server/protocol.py

```diff
@@ -399,7 +390,7 @@ def get_extension_statuses(cls, slices: list["ServerSlice"]) -> list[ExtensionSt
                 result[ext_status.name] = ext_status
         return list(result.values())
 
-    async def get_status(self) -> dict[str, ArgumentTypes]:
+    async def get_status(self) -> Mapping[str, ArgumentTypes | Mapping[str, ArgumentTypes]]:
```

Comment:
Why change all these from `dict` to `Mapping`?

Reply:
Due to invariance of dict, the Union is a bit hard to use otherwise  and it is inteded to be mapping all along

Reply:
Left it like this, I think Mapping better reflects the intent.

Reply:
I don't necessarily agree as a general rule, but I can see your point for this case.